service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -177,20 +181,36 @@ public ResponseEntity<PostMethodResponse> postMethod(PostMethodRequest postMetho
       UUID methodId = UUID.randomUUID();
       UUID runSetId = UUID.randomUUID();
 
+      GithubUrlComponents githubUrlComponents =
+          MethodUtil.extractGithubDetailsFromUrl(rawMethodUrl);
+
+      String path = githubUrlComponents.path();
+      String repository = githubUrlComponents.repo();
+      String organization = githubUrlComponents.org();
+      String branchOrTagName = githubUrlComponents.branchOrTag();
+
+      GithubMethodSource githubMethodSource =
```

Comment:
Similar to another comment I made -- it feels to me like `GithubMethodDetails` is more accurate.

I saw you made a comment elsewhere arguing that using the word "Source" prepares us for adding sources beyond GitHub, but I think we could get that clarity with something like `DockerhubMethodDetails` or `SomeothersourceMethodDetails`